network-api/networkapi/wagtailpages/pagemodels/campaigns.py

```diff
@@ -426,3 +450,16 @@ def get_context(self, request):
     class Meta:
         verbose_name = "Banner Page"
         verbose_name_plural = "Banner pages"
+
+    def clean(self):
+        if self.use_intro_from_parent and self.intro:
+            raise ValidationError({
+                'use_intro_from_parent': ValidationError("Please use only one intro option."),
+                'intro': ValidationError("Please use only one intro option.")
+            })
+        if self.use_intro_from_parent:
+            parent = self.get_parent()
+            if not hasattr(parent.specific, 'intro'):
+                raise ValidationError({
+                    'use_intro_from_parent': ValidationError("Parent intro field does not exist"),
+                })
```

Comment:
Just noticing this can fail when the page is first created, because there is no parent that can be retrieved at that point.

Reply:
Maybe I can do something with a  custom form. [https://docs.wagtail.org/en/stable/advanced_topics/customisation/page_editing_interface.html#custom-edit-handler-forms](https://docs.wagtail.org/en/stable/advanced_topics/customisation/page_editing_interface.html#custom-edit-handler-forms)

I probably want to add a test for this as well.